internal/provider/resource/identity_kubernetes_auth.go

```diff
@@ -197,6 +229,83 @@ func updateKubernetesAuthStateByApi(ctx context.Context, diagnose diag.Diagnosti
 	plan.AccessTokenTrustedIps = stateAccessTokenTrustedIps
 }
 
+func validateTokenReviewerMode(diagnose *diag.Diagnostics, plan *IdentityKubernetesAuthResourceModel) {
+
+	if plan == nil {
+		diagnose.AddError(
+			"Invalid plan",
+			"Plan is nil",
+		)
+		return
+	}
+
+	tokenReviewerMode := plan.TokenReviewerMode.ValueString()
+
+	if tokenReviewerMode == TOKEN_REVIEWER_MODE_GATEWAY {
+		if plan.GatewayID.ValueString() == "" {
+			diagnose.AddError(
+				"Gateway ID is required",
+				"Gateway ID is required when token reviewer mode is gateway. Please provide a valid gateway ID.",
+			)
+			return
+		}
+
+		if plan.CaCertificate.ValueString() != "" {
+			diagnose.AddError(
+				"Cannot set CA certificate",
+				"CA certificate is not allowed when token reviewer mode is gateway. Please remove the CA certificate.",
+			)
+			return
+		}
+
+		if plan.KubernetesHost.ValueString() != "" {
+			diagnose.AddError(
+				"Cannot set Kubernetes host",
+				"Kubernetes host is not allowed when token reviewer mode is gateway. Please remove the Kubernetes host.",
+			)
+			return
+		}
+
+		if plan.TokenReviewerJWT.ValueString() != "" {
+			diagnose.AddError(
+				"Cannot set Token reviewer JWT",
+				"Token reviewer JWT is not allowed when token reviewer mode is gateway. Please remove the Token reviewer JWT.",
+			)
+			return
+		}
+	} else if tokenReviewerMode == TOKEN_REVIEWER_MODE_API {
+		if plan.GatewayID.ValueString() != "" {
+			diagnose.AddError(
+				"Cannot set gateway ID",
+				"Gateway ID is not allowed when token reviewer mode is api. Please remove the gateway ID.",
+			)
+			return
+		}
+
+		if plan.TokenReviewerJWT.ValueString() == "" {
+			diagnose.AddError(
+				"Token reviewer JWT is required",
+				"Token reviewer JWT is required when token reviewer mode is api. Please provide a valid token reviewer JWT.",
+			)
+			return
+		}
```

Comment:
also, isn't this optional because we do self-review without this? 

<img width="833" height="565" alt="image" src="https://github.com/user-attachments/assets/4f299c68-f176-4f2b-9591-e908a7481492" />